pros/src/motor.rs

```diff
@@ -274,6 +279,25 @@ impl From<i32> for Gearset {
     }
 }
 
+pub struct MotorStoppedFuture {
+    motor: Motor,
+}
+impl core::future::Future for MotorStoppedFuture {
+    type Output = crate::Result;
+    fn poll(
+        self: core::pin::Pin<&mut Self>,
+        cx: &mut core::task::Context<'_>,
+    ) -> core::task::Poll<Self::Output> {
+        match self.motor.get_state()?.stopped {
+            true => core::task::Poll::Ready(Ok(())),
+            false => {
+                cx.waker().wake_by_ref();
+                core::task::Poll::Pending
+            }
+        }
```

Comment:
This seems just like an if statement but with more steps

Reply:
you could also write it with an if, but a match works just fine. I specifically chose a match here because I thought it looked better.